planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/utils.cpp

```diff
@@ -311,6 +325,19 @@ bool set_goal(
         }
       }
 
+      // pre refined goal
+      auto & pre_mid_goal = output_ptr->points.at(output_ptr->points.size() - 2);
+      for (size_t i = min_dist_out_of_circle_index + 1; i < input.points.size(); ++i) {
+        for (const auto target_lane_id : input.points.at(i).lane_ids) {
+          const bool is_lane_id_found =
+            std::find(pre_mid_goal.lane_ids.begin(), pre_mid_goal.lane_ids.end(), target_lane_id) !=
+            pre_mid_goal.lane_ids.end();
+          if (!is_lane_id_found) {
+            pre_mid_goal.lane_ids.push_back(target_lane_id);
+          }
+        }
+      }
+
```

Comment:
minor:
The logic is nearly the same. The only difference is whether the value is -1 or -0.5.We could generalize this with a lambda function.